common/txmgr/types/client.go

```diff
@@ -69,11 +71,31 @@ type TxmClient[
 	R ChainReceipt[TX_HASH, BLOCK_HASH],
 	SEQ Sequence,
 	FEE feetypes.Fee,
+] interface {
+	TransactionClient[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]
+
+	// receipt fetching used by confirmer
+	BatchGetReceipts(
+		ctx context.Context,
+		attempts []TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
+	) (txReceipt []R, txErr []error, err error)
+}
+
+// TransactionClient contains the methods for building, simulating, broadcasting transactions
+type TransactionClient[
+	CHAIN_ID ID,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	SEQ Sequence,
+	FEE feetypes.Fee,
 ] interface {
 	ConfiguredChainID() CHAIN_ID
+	PendingSequenceAt(ctx context.Context, addr ADDR) (SEQ, error)
+	SequenceAt(ctx context.Context, addr ADDR, blockNum *big.Int) (SEQ, error)
```

Comment:
my thinking was that the nonces were used to build the tx, but the updating the nonce is not part of transaction management

but i think it could organized a bit better - I'll take another look